examples/responsive-vis/responsive-bar-chart.js

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import React from 'react';
+
+import {
+  AreaSeries,
+  Hint,
+  HorizontalBarSeries,
+  XAxis,
+  XYPlot,
+  YAxis
+} from 'react-vis';
+
+import {
+  filterFeatures,
+  getPPP
+} from './responsive-vis-utils';
+
+// range constants
+const VERY_LOW_RANGE = [0, 0.08];
+const LOW_RANGE = [0, 0.7];
+const VERY_LOW_TO_HIGH = [0.08, Infinity];
+const HIGH_RANGE = [0.7, Infinity];
+// const MIN_PPP = 2e-5;
+
+export const BARCHART_FEATURES = [
+  {min: -Infinity, max: Infinity, name: 'xaxis', group: 0},
+  {min: VERY_LOW_RANGE[0], max: VERY_LOW_RANGE[1], name: 'yaxis', group: 1},
+  {min: VERY_LOW_TO_HIGH[0], max: VERY_LOW_TO_HIGH[1], name: 'mouseLabels', group: 1},
+  {min: LOW_RANGE[0], max: LOW_RANGE[1], name: 'bars', group: 2},
+  {min: HIGH_RANGE[0], max: HIGH_RANGE[1], name: 'area', group: 2}
+];
+
+function updateDataForArea(data, ppp) {
+  // Use the PPP ratio as the step to sample the data
+  const step = Math.round(ppp);
+  const sample = [];
```

Comment:
Do you think we could map this sample by creating an Array of size corresponding to the floored division of `data.length` by `step`?

Something like:

```
const updateDataForArea = (data, ppp) =>
  Array(Math.floor(data.length / Math.round(ppp))).fill(0)
    .map((_, index) => ({...data[index * Math.round(ppp)], y: index}))
```

⚠️  didn't test, and maybe I'm trying to hard 😝